### ISSUE TITLE
Deadlock during lane change of 2 vehicles 1.13.0 version

### ISSUE THREAD

**Issue by lola-notraffic** [eclipse/sumo#12028]
Hi,
While changing lanes: vehicle changing to right lane vs. vehicle changing to left lane (as circled in the attached image) - deadlock happened.
There are different parameters in LaneChange that may be defined to improve it. We tried some of them, which changed some of atspm metrics that we measured  drastically. We also ran 1.15 SUMO version and deadlock did not  occur. So we have a few questions:
1. What changed from 1.13 to 1.15 version in terms of deadlock while vehicles changing lanes?
2. Because of a drastic change in atspm parameters calculated before and after change in lane change parameters for vehicles:
what do you suggest the best parameter from Lane Change to use to solve the deadlock and to cause least changes to simulation performance (we still need to use 1.13 version, so we have to find solution for this specific version).

![image (11)](https://user-images.githubusercontent.com/95179338/201517085-5b5f6454-f9c6-47ff-9981-6e729728bc8c.png)

Thank you in advance,
Olga

**SUMO-version:*1.13.0*

**operating system:*ubuntu 20.04*

**Comment by namdre** [comment]
There were no specific bugfixes with regard to lane-changing deadlock after version 1.13.0 (while 1.13.0 actually had such fixes). However, some of the more recent bugfixes could plausibly cause the improved behavior as a side effect. (see https://sumo.dlr.de/docs/ChangeLog.html).
Unless you have run many variations of your scenario (with different inputs or random seeds), the difference in measures might also be due to noise.

That being said, the situation in your screenshot does rather look like a bug to me so it's not something you can prevent by jus tuning some parameters.  What you can do is avoid cars being in such a "weaving" situation:
- define vehicles with attribute `departLane="best"` 
- let turning movements at intersections target additional lanes so that cars reach the desired target lane without lane changing (but only if there are no conflicts with other connections permitted in the same phase).

Questions from my end:
- which ATSPM parameters do you calculate?
- What prevents you from upgrading to a new version? We generally aim to provide options for behavioral backward compatibility (though nothing of the sort should be needed between 1.13.0 and 1.15.0) so that the only changes you should be seeing are due to bugfixes.

If you can make any changes to your sumo version at all, you could also try to pinpoint the specific fix that was responsible for the deadlocks and only backport that patch for your version (git-bisect makes it somewhat simply to do so).

**Comment by lola-notraffic** [comment]
Hi again,
As you suggested I also ran the option departLane="best". But somehow it did not help in our case. Actually increasing lcLookaheadLeft value from default helped, but we are still not sure about this change, because it changed some performances/atspms.
About your questions on atspms: we calculate arrivals on green and avg delay. We had same scenario where avg delay we got was reasonable (version 1.13), but while running version 1.15 - avg delay dropped drastically (example: from 17 sec (in version 1.13) for one of left turns to 2 secs (version 1.15), in other directions it also decreased). We will continue to check if this change makes sense to what we expect.
